ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -189,30 +190,23 @@ public void handleGet(final RestRequest restRequest, final RestResponseChannel r
         GetCallback routerCallback = new GetCallback(restRequest, restResponseChannel, subresource, options);
         SecurityProcessRequestCallback securityCallback =
             new SecurityProcessRequestCallback(restRequest, restResponseChannel, routerCallback);
-        RestRequestMetrics requestMetrics =
-            restRequest.getSSLSession() != null ? frontendMetrics.getBlobSSLMetrics : frontendMetrics.getBlobMetrics;
         if (subresource != null) {
           logger.trace("Sub-resource requested: {}", subresource);
-          switch (subresource) {
-            case BlobInfo:
-              requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getBlobInfoSSLMetrics
-                  : frontendMetrics.getBlobInfoMetrics;
-              break;
-            case UserMetadata:
-              requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getUserMetadataSSLMetrics
-                  : frontendMetrics.getUserMetadataMetrics;
-              break;
-            case Replicas:
-              requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getReplicasSSLMetrics
-                  : frontendMetrics.getReplicasMetrics;
-              securityCallback = new SecurityProcessRequestCallback(restRequest, restResponseChannel);
-              break;
+          if (subresource == SubResource.Replicas) {
+            securityCallback = new SecurityProcessRequestCallback(restRequest, restResponseChannel);
           }
         }
-        restRequest.getMetricsTracker().injectMetrics(requestMetrics);
+        boolean isEncrypted = new BlobId(operationOrBlobId, clusterMap).isEncrypted();
```

Comment:
thanks for pointing it out. I will create BlobId the same way as you suggested, but will leave injectTargetAccountAndContainerFromBlobId() as is. Don't wanna change it to take in BlobId. If not, have to change InBoundIdConverterCallback() impl or api which is uncalled for considering this patch.

Reply:
Not sure I understand. You can have to create the `BlobId` in `InboundIdConverterCallback` - so what's the extra work required to make `AccountAndContainerInjector#injectTargetAccountAndContainerFromBlobId()` take in a `BlobId`? You can't generate a `BlobId` at this point anyway

Reply:
sorry, why do I have to create BlobId in InboundIdConverterCallback?

Reply:
As mentioned in my original comment it is because it is possible that the ID at this point is not a valid Ambry ID since the ID converter has not yet run